### PR TITLE
add typed assignment functions to support dynamic configuration (FF-803)

### PR DESCRIPTION
**Pull request by leoromanovsky**
## motivation

Currently always returning a `String` for `getAssignment`

## description

* adds new typed `get*Assignment` methods for `bool`, `double` and `string`
* enables github unit tests by creating a `testTarget`
* delete test data stored in git; keeps consistent with other repos
* version bump to 1.0.0

### CODE REVIEW

**Review by evanmiller — APPROVED** [review]

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
+name: Swift
 
-# on: [push]
+on: [push]
 
-# jobs:
-#   build:
+jobs:
+  build:
+    runs-on: macos-latest
```
Comment (aarsilv) [inline .github/workflows/unit-tests.yml:7]: 🤯 
I remember when macos was a no-go for the cloud and we had a stack of mac minis in the office with a sticky note on their UPS saying DO NOT UNPLUG

File: Sources/eppo/EppoClient.swift [hunk 2]
```diff
@@ -45,14 +45,59 @@ public class EppoClient {
...
+        return try getAssignment(subjectKey, flagKey, SubjectAttributes())?.boolValue()
+    }
+    
+    public func getBoolAssignment(
+        _ subjectKey: String,
+        _ flagKey: String,
+        _ subjectAttributes: SubjectAttributes) throws -> Bool?
+    {
+        return try getAssignment(subjectKey, flagKey, subjectAttributes)?.boolValue()
+    }
```
Comment (aarsilv) [inline Sources/eppo/EppoClient.swift:61]: I think we can collapse these into one function declaration [leveraging default parameter values](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/functions#Default-Parameter-Values).

```swift
public func getBoolAssignment(
    _ subjectKey: String,
    _ flagKey: String,
    _ subjectAttributes: SubjectAttributes = SubjectAttributes()) throws -> Bool?
{
    return try getAssignment(subjectKey, flagKey, subjectAttributes)?.boolValue()
}
```

Same for the other types below

File: Sources/eppo/EppoClient.swift [hunk 3]
```diff
@@ -45,14 +45,59 @@ public class EppoClient {
...
+    {
+        return try getAssignment(subjectKey, flagKey, subjectAttributes)?.stringValue()
+    }
+    
+    private func getAssignment(_ subjectKey: String, _ flagKey: String) throws -> EppoValue? {
         return try getAssignment(subjectKey, flagKey, [:]);
     }
 
-    public func getAssignment(
+    private func getAssignment(
```
Comment (aarsilv) [inline Sources/eppo/EppoClient.swift:97]: By making this private, could we be breaking things for SDK users not explicitly using the typed value methods yet?

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (leoromanovsky) [reply .github/workflows/unit-tests.yml:7]: Now it's an enclosure around a Mac mini 🤣

File: Sources/eppo/EppoClient.swift (hunk 3 above)
Comment (leoromanovsky) [reply Sources/eppo/EppoClient.swift:97]: Yes this is a breaking change and we will set the version number to a "major" bump.

**Pull request merged**